### ISSUE TITLE
Collecting an array of identical refs

### ISSUE THREAD

**Issue by mikehwagz** [terkelg/facon#1]
Loving this clever little utility!

After experimenting a bit, I've discovered that I can't easily store an array of refs. Unfortunately, if the same ref is used more than once, the element with the last occurrence of the ref will be stored.

I came up with a quick solution where any identical refs result in an array of the elements:

```js
  content.collect = ({ attr = 'ref', keepAttribute, assign = {} } = {}) => {
    const refElements = content.querySelectorAll(`[${attr}]`)
    return [...refElements].reduce((acc, element) => {
      const propName = element.getAttribute(attr).trim()
      !keepAttribute && element.removeAttribute(attr)
      return {
        ...acc,
        [propName]: acc[propName]
          ? Array.isArray(acc[propName])
            ? [...acc[propName], element]
            : [acc[propName], element]
          : element,
      }
    }, assign)
  }
```

What do you think?

**Comment by terkelg** [comment]
Hi, Mike 👋 Thanks for the suggestion.
Looks good to me. My only concern is the browser support for the object spread operator.
Do you want to do a PR with updated docs and tests? Could be awesome! 💯

**Comment by mikehwagz** [comment]
Good catch! I'll work on an implementation without the object spread operator and submit a PR. 

Cheers 🍻

**Comment by terkelg** [comment]
Published in v1.0.2 🎉
